src/components/AddBoxForm/AddBoxFormMobile.jsx

```diff
@@ -0,0 +1,247 @@
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import DatePicker from 'react-datepicker';
+import { useForm, Controller } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+
+import {
```

Comment:
All the yup-related variables need to be instantiated in the component its being used in from what I've seen so it wouldn't work to have the errors variable in a useState in AddBoxForm since it's coming from yup. Also, in general, we don't need to worry too much about any changes when switching between mobile and desktop views since that wouldn't happen unless the user is drastically changing screen sizes.

Reply:
Besides the errors, I think the re-rendering of the form from desktop to mobile also clears whatever the user had typed in the boxes beforehand, so we can properly save the form data using useState as well(?). I am not sure if that would be important, but I want to point that out before we approve.

Reply:
Refactored add box form so it is just one component now instead of two separate components for mobile and desktop views. This should fix this issue, but please test on your end as well